add notification docs (#73)
MiSArch/docs: docs/docs/dev-manuals/services/notification.mdx

- **Database**: PostgreSQL
- **Schema Management**: Flyway

## Repository Structure

The repository is structured as follows:

<div className="repository-structure" >

- `/src/main/`: Source code of the service
  - `kotlin/org/misarch/notification/`: Kotlin source code
    - `event/`: Dapr event handling
      - `model/`: Event DTOs
    - `graphql/`: GraphQL API
      - `dataloader/`: Data loaders for efficient data fetching
      - `federation/`: Apollo Federation entitiy resolvers
      - `input/`: Input types for mutations
      - `model/`: GraphQL types
        - `connection/`: Connection types and logic for paginated queries
    - `persistance/`: Database entities and repositories
      - `model/`: R2DBC entities
      - `repository/`: Spring Data R2DBC repositories
    - `service/`: Service logic
  - `resources/`: Spring Boot application configuration and SQL schema migration files
    - `db/migration/`: Flyway SQL schema migration files

</div>

## Defined Events

This section lists events that are defined by the notification service and can be used by other services.

### Published Defined Events

n/a

### Subscribed Defined Events

#### notification/notification/create

> This event can be used by other services to create new notifications.

Reviewer: ```suggestion
Other services can publish this event to create new notifications.
```

```js
{
  "title": "string",
  "body": "string",
  "userId": "boolean"
}
```

## Imported Events

This section lists events defined by other services that are used by the notification service.

### Published Imported Events

n/a

### Subscribed Imported Events

> - [user/user/created](/docs/docs/dev-manuals/services/user#userusercreated): Used to import `User` from the `User` bounded context.

Reviewer: ```suggestion
- [user/user/created](/docs/docs/dev-manuals/services/user#userusercreated): Used to validate that only existing users can receive notifications
```
